src/d/actor/d_a_npc_bs1.cpp

```diff
@@ -137,161 +567,651 @@ void daNpc_Bs1_c::shop_talk() {
 }
 
 /* 00003478-000034C8       .text setAttention__11daNpc_Bs1_cFb */
-void daNpc_Bs1_c::setAttention(bool) {
+void daNpc_Bs1_c::setAttention(bool shouldSet) {
     /* Nonmatching */
+    if (!shouldSet && m72F > 1) {
+        return;
+    }
+    // Hard to get this ordered correctly. At 70% match, good enough for now.
+    f32 zPos = m718.z;
+    f32 yPos = l_HIO.mpChildArray[mLHioChildIdx].mHIO.mAttnYOffset;
+    attention_info.position.y = m718.y + yPos;
+    attention_info.position.x = m718.x;
+    attention_info.position.z = zPos;
 }
 
 /* 000034C8-000037B8       .text lookBack__11daNpc_Bs1_cFv */
 void daNpc_Bs1_c::lookBack() {
     /* Nonmatching */
+    cXyz destPos;
+    cXyz other;
+    f32 desiredYRot = current.angle.y;
+    switch (m830) {
+    case 0x2:
+        if (mShopCamAction.mCurrActionFunc == NULL) {
+            if (mShopItems.mSelectedItemIdx == -1) {
+                // What?
+                mShopCamAction.getItemZoomPos(100.0f);
+            } else {
+                cXyz selectItemBasePos = mShopItems.getSelectItemBasePos();
+                cXyz selectItemPos = mShopItems.getSelectItemPos();
+                mpShopCursor->setPos(selectItemBasePos);
+                daNpc_Bs1_childHIO_c child = l_HIO.mpChildArray[mLHioChildIdx];
+                mpShopCursor->m48 = child.m30;
+                mpShopCursor->m4C = child.m34;
+                mpShopCursor->m50 = child.m38;
+                mpShopCursor->m38 = child.m3C;
+                mpShopCursor->m3C = child.m40;
+                mpShopCursor->anm_play();
+            }
+        } else {
+            destPos = dNpc_playerEyePos(0.0f);
+        }
+        other = cXyz(current.pos.x, eyePos.y, current.pos.z);
+        break;
+    case 0x1:
+        if (m730 == 0) {
+            destPos = cXyz();
+            setTexAnm(ANM_UNK_1);
+        } else {
+            destPos = dNpc_playerEyePos(0.0f);
+            other = cXyz(current.pos.x, eyePos.y - 80.0f, current.pos.z);
+            setTexAnm(ANM_UNK_2);
+        }
+    }
+    if (mJntCtrl.mbTrn == false) {
+        m724 = 0;
+    } else {
+        cLib_addCalcAngleS2(&m724, l_HIO.mpChildArray[mLHioChildIdx].mHIO.mMaxAttnAngleY, (s16)4,
+                            (s16)0x800);
+    }
+    mJntCtrl.lookAtTarget(&current.angle.y, &destPos, other, desiredYRot, m724, true);
 }
 
 /* 000037B8-00003830       .text wait01__11daNpc_Bs1_cFv */
-void daNpc_Bs1_c::wait01() {
+BOOL daNpc_Bs1_c::wait01() {
     /* Nonmatching */
+    if (m731 != 0) {
+        m831 = m830;
+        m830 = 2;
+    } else {
+        if (m730 != 0) {
+            m82A = 2;
+            // cmplwi but is cmpwi
+        } else if (m82A == 0x02) {
+            m82A = 0;
+        }
+    }
+    // Return is also not matching
+    return mpMorf->mCurMorf < 1.0f;
 }
 
 /* 00003830-00003970       .text talk01__11daNpc_Bs1_cFv */
-void daNpc_Bs1_c::talk01() {
-    /* Nonmatching */
+BOOL daNpc_Bs1_c::talk01() {
+    u16 value = talk();
```

Comment:
```suggestion
    u16 status = talk();
```
the u16 returned by npc talk functions is the message status